workflows/embedded/index.ts

```diff
@@ -0,0 +1,258 @@
+import { recordHook } from '@flatfile/plugin-record-hook';
+import api from '@flatfile/api';
+import { ExcelExtractor } from '@flatfile/plugin-xlsx-extractor';
+import { pushToHcmShow } from '../../actions/pushToHCMShow';
+import { blueprintSheets } from '../../blueprints/benefitsBlueprint';
+import { benefitValidations } from '../../recordHooks/benefits/benefitElectionsValidations';
+
+type Metadata = {
+  userId: string;
+};
+
+// Define the main function that sets up the listener
+export default function (listener) {
+  // Log the event topic for all events
+  listener.on('**', (event) => {
+    console.log('> event.topic: ' + event.topic);
+  });
+
+  // Add an event listener for the 'job:created' event
+  listener.filter({ job: 'space:configure' }, (configure) => {
+    configure.on('job:ready', async (event) => {
+      console.log('Reached the job:ready event callback');
+
+      // Destructure the 'context' object from the event object to get the necessary IDs
+      const { spaceId, environmentId, jobId } = event.context;
+
+      const space = await api.spaces.get(spaceId);
+
+      console.log('Space: ' + JSON.stringify(space));
+
+      const metadata = space.data.metadata as Metadata;
+
+      const userId = metadata.userId;
+
+      const updateJob1 = await api.jobs.ack(jobId, {
+        info: 'Creating Space',
+        progress: 10,
+      });
+
+      console.log('Updated Job: ' + JSON.stringify(updateJob1));
+
+      // Log the environment ID to the console
+      console.log('env: ' + environmentId);
+      console.log('spaceId ' + spaceId);
+      console.log('jobID: ' + jobId);
+
+      try {
+        // Create a new workbook using the Flatfile API
+        const createWorkbook = await api.workbooks.create({
+          spaceId: spaceId,
+          environmentId: environmentId,
+          labels: ['primary'],
+          name: 'Benefits Workbook',
+          sheets: blueprintSheets,
+          actions: [
+            {
+              operation: 'submitAction',
+              slug: 'BenefitsWorkbookSubmitAction',
```

Comment:
Does this action work in HCM? If not may want to comment out for now. If it does, this slug doesn't match what we're checking for below